pkg/schema/schema.go

```diff
@@ -97,8 +97,14 @@ func NewMapItemType(item *yamlmeta.MapItem) (*MapItemType, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	return &MapItemType{Key: item.Key, ValueType: typeOfValue, defaultValue: typeOfValue.GetDefaultValue(), Position: item.Position}, nil
+	defaultValue, err := getExplicitDefaultValue(item)
+	if err != nil {
+		return nil, err
+	}
+	if defaultValue == nil {
```

Comment:
what if the explicit default is actually specified to be null?

Reply:
Updated to allow null.